added file tree theme and search

jstucke opened the pull request:
- added bootstrap jstree theme for a more intuitive interface 
- added search plugin and search field for searching in the file tree (only works on expanded nodes because of lazy loading)

dorpvom reviewed (commented):
This works, it looks good and seems helpful. On larger firmware (> 2000 files) the search becomes laggy on my system which notably reduces the usability. Also I found it un-intuitive, that one has to first expand the tree to inner containers to allow searching in those. This latter thing might not be solvable but we at least might want to add sth. like a tooltip to help on a red error bar (~ "Did you already expand  the tree to the container / file system you want to search?")

jstucke commented:
> This works, it looks good and seems helpful. On larger firmware (> 2000 files) the search becomes laggy on my system which notably reduces the usability. Also I found it un-intuitive, that one has to first expand the tree to inner containers to allow searching in those. This latter thing might not be solvable but we at least might want to add sth. like a tooltip to help on a red error bar (~ "Did you already expand the tree to the container / file system you want to search?")

How about this: 
- we add a search button and don't search on any key press (-> less laggy)
- we automatically fully expand the tree when initiating the search (could be complicated but shouldn't take that long)

maringuu commented:
> we add a search button and don't search on every key press (-> less laggy)

Maybe this is a bit overcomplicated but what about automatically seraching after some time passed? E.g. 0.2 Seconds.

jstucke commented:
- merged master into the branch and added the theme package to npm dependencies
- added a delay to the search
- added some user feedback in case there are no matches

maringuu commented:
The code LGTM besides the missing newline. Somehow the network does not allow me to install the node package.

jstucke commented on src/web_interface/static/js/show_analysis_fileTreeAjax.js:
```diff
@@ -7,9 +7,43 @@ $('#fileTreeAjax')
+        $('#fileTreeAjax').jstree('search', $(this).val());
+        $('#fileTreeAjax .jstree-hidden').hide();
+    }, 1000));
+});
```
fixed
